cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationsDaoImpl.java

```diff
@@ -56,70 +58,49 @@ public String getLocations(String names,String format, String units, String datu
-        String[] stateInitial = new String[1];
-        Boolean[] active = new Boolean[1];
-        String[] locationKindId = new String[1];
-        String[] mapLabel = new String[1];
-        Number[] publishedLatitude = new Number[1];
-        Number[] publishedLongitude = new Number[1];
-        String[] boundingOfficeId = new String[1];
-        String[] nationId = new String[1];
-        String[] nearestCity = new String[1];
-        ResultSet[] aliasResultSet = new ResultSet[1];
-        AtomicReference<Location> locationRef = new AtomicReference<>();
-        try {
-            dsl.connection(c ->
-            {
-                CwmsDbLoc locJooq = CwmsDbServiceLookup.buildCwmsDb(CwmsDbLoc.class, c);
-                locJooq.retrieve2(c, officeId, locationId, elevationUnitId, locationType, elevation, verticalDatum, latitude, longitude,
-                        horizontalDatum, publicName, longName, description, timezoneId, countyName, stateInitial, active, locationKindId,
-                        mapLabel, publishedLatitude, publishedLongitude, boundingOfficeId, nationId, nearestCity, aliasResultSet);
-                Location location = new Location.Builder(locationId[0], locationKindId[0], ZoneId.of(timezoneId[0]), latitude[0].doubleValue(), longitude[0].doubleValue(), horizontalDatum[0], officeId)
-                        .withLocationType(locationType[0])
-                        .withElevation(elevation[0].doubleValue())
-                        .withVerticalDatum(verticalDatum[0])
-                        .withPublicName(publicName[0])
-                        .withLongName(longName[0])
-                        .withDescription(description[0])
-                        .withCountyName(countyName[0])
-                        .withStateInitial(stateInitial[0])
-                        .withActive(active[0])
-                        .withMapLabel(mapLabel[0])
-                        .withBoundingOfficeId(boundingOfficeId[0])
-                        .withNearestCity(nearestCity[0])
-                        .withNation(Nation.NationForName(nationId[0]))
-                        .build();
-                if (publishedLatitude[0] != null) {
-                    location = new Location.Builder(location)
-                            .withPublishedLatitude(publishedLatitude[0].doubleValue())
-                            .build();
-                }
-                if (publishedLongitude[0] != null) {
-                    location = new Location.Builder(location)
-                            .withPublishedLongitude(publishedLongitude[0].doubleValue())
-                            .build();
-                }
-                locationRef.set(location);
-            });
-        }
-        catch(DataAccessException ex)
-        {
-            throw new IOException("Could not retrieve Location");
-        }
-        return locationRef.get();
+
+        SelectQuery<Record> query = dsl.selectQuery();
```

Comment:
This one down purely to allow for the Mock. the current version of JOOQ we're using doesn't have a MockDataProvider that handles the appropriate JSON data. Might be worth creating our own, might now. No super worried about it at the moment.